pkg/internal/storage/repo/transactions.go

```diff
@@ -485,3 +554,55 @@ func (txs *Transactions) labelFilterScope(tx *gorm.DB, userID int, filter entity
 		return query.Where("id IN (?)", subQuery)
 	}
 }
+
+func (txs *Transactions) AbortTransactionAtomic(ctx context.Context, transactionID uint, txID *string, reference string) error {
+	if err := txs.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
+		txNotes := history.NewBuilder().AbortAction(reference).Note().AsList()
+		historyBuilders := make([]history.Builder, len(txNotes))
+		for i, note := range txNotes {
+			historyBuilders[i] = history.NewBuilderFromNote(note)
+		}
```

Comment:
Ohh, thanks to sonar, I just noticed that:
1. you create a builder
2. asked builder to create a note and create a slice with this one note
3. iterate over this slice with one note
4. create a builder from this one note
5. add a builder to a new slice

So it could be simplified and optimised by:
```suggestion
		historyBuilders := []history.Builder{history.NewBuilder().AbortAction(reference)}
```

Reply:
simplified